src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/ManagedIndexRunner.kt

```diff
@@ -220,26 +222,26 @@ object ManagedIndexRunner : ScheduledJobRunner,
     }
 
     private suspend fun initializeManagedIndexMetaData(managedIndexConfig: ManagedIndexConfig, policy: Policy?) {
+        val stateMetaData = if (policy?.defaultState != null) {
+            StateMetaData(policy.defaultState, Instant.now().toEpochMilli())
+        } else {
+            null
+        }
         val managedIndexMetaData = ManagedIndexMetaData(
             index = managedIndexConfig.index,
             indexUuid = managedIndexConfig.indexUuid,
-            policyName = managedIndexConfig.policyName,
+            policyID = managedIndexConfig.policyID,
             policySeqNo = policy?.seqNo,
             policyPrimaryTerm = policy?.primaryTerm,
-            state = policy?.defaultState,
-            stateStartTime = if (policy != null) Instant.now().toEpochMilli() else null,
-            transitionTo = null,
-            actionIndex = null,
-            action = null,
-            actionStartTime = null,
-            step = null,
-            stepStartTime = null,
-            stepCompleted = null,
-            failed = policy == null,
             policyCompleted = false,
             rolledOver = false,
-            info = if (policy == null) mapOf("message" to "Could not load policy: ${managedIndexConfig.policyName}") else null,
-            consumedRetries = null
+            transitionTo = null,
+            stateMetaData = stateMetaData,
+            actionMetaData = null,
+            stepMetaData = null,
+            // TODO fix retryInfo when we implement retry logic.
+            retryInfo = RetryInfoMetaData(failed = policy == null, consumedRetries = 0),
+            info = if (policy == null) mapOf("message" to "Could not load policy: ${managedIndexConfig.policyID}") else null
```

Comment:
If there is a policy, we could add a message that we successfully initialized or loaded policy